aci/data_source_aci_vnsldevvip.go

```diff
@@ -0,0 +1,158 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAciL4ToL7Devices() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceAciL4ToL7DevicesRead,
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"tenant_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"active": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"annotation": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"context_aware": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"device_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"function_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"is_copy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"managed": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"promiscuous_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"service_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"trunking": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"relation_vns_rs_al_dev_to_dom_p": {
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Description:   "Create relation to vmmDomP",
+				MaxItems:      1,
+				ConflictsWith: []string{"relation_vns_rs_al_dev_to_phys_dom_p"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"domain_dn": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
This one shouldn't be required in the datasource right? Should be optional and computed right?

Reply:
Changed it to optional